levels.go

```diff
@@ -34,12 +37,147 @@ import (
 	"golang.org/x/time/rate"
 )
 
+const inactiveEpoch = math.MaxUint64
+
+type tableRefManager struct {
+	currentEpoch uint64
+
+	mu             sync.Mutex
+	guards         []*tableRef
+	inactiveGuards []*tableRef
+}
+
+func newTableRefManager(c *y.Closer) *tableRefManager {
+	e := new(tableRefManager)
+	go e.gcLoop(c)
+	return e
+}
+
+func (e *tableRefManager) refTables() *tableRef {
+	g := &tableRef{
+		epoch:      e,
+		localEpoch: atomic.LoadUint64(&e.currentEpoch),
+	}
+	e.mu.Lock()
+	e.guards = append(e.guards, g)
+	e.mu.Unlock()
+	return g
+}
+
+func (e *tableRefManager) gcLoop(c *y.Closer) {
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+	defer c.Done()
+
+	for {
+		select {
+		case <-ticker.C:
+			e.doGC()
+		case <-c.HasBeenClosed():
+			e.clear()
+			return
+		}
+	}
+}
+
+func (e *tableRefManager) clear() {
```

Comment:
The next reopen will delete all the unreachable tables, we don't need to clear them before close.